app/src/main/java/com/example/calculatordemo/MainActivity.kt

```diff
@@ -1,12 +1,282 @@
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+
 
 class MainActivity : AppCompatActivity() {
+    private var mFirstValue: Float = 0.0F
+    private var mSecondValue: Float = 0.0F
+    private var mResult: Float = 0.0F
+    private var mTimesChangeTheme: Int = 0
+    private lateinit var mTextViewDisplay: TextView
+
+    private lateinit var mButtonOne: Button
+    private lateinit var mButtonTwo: Button
+    private lateinit var mButtonThree: Button
+    private lateinit var mButtonFour: Button
+    private lateinit var mButtonFive: Button
+    private lateinit var mButtonSix: Button
+    private lateinit var mButtonSeven: Button
+    private lateinit var mButtonEight: Button
+    private lateinit var mButtonNine: Button
+    private lateinit var mButtonZero: Button
+
+    private lateinit var mButtonSum: Button
+    private var mSum: Boolean = false
+    private lateinit var mButtonSubstract: Button
+    private var mSubstract: Boolean = false
+    private lateinit var mButtonMultiply: Button
+    private var mMultiply: Boolean = false
+    private lateinit var mButtonDivide: Button
+    private var mDivide: Boolean = false
+    private lateinit var mButtonResult: Button
+
+    private lateinit var mButtonComma: Button
+    private lateinit var mButtonInterest: Button
+    private lateinit var mButtonNegative: Button
+
+    private lateinit var mButtonReset: Button
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        //initialize the variables
+        initVariables()
+        //print the saved result
+        printLastSavedResult()
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        onClickButton()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        val inflater: MenuInflater = menuInflater
+        inflater.inflate(R.menu.main_menu, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
+        return when (item?.itemId) {
+            R.id.menu_change_theme -> {
+                mTimesChangeTheme++
+                if (mTimesChangeTheme % 2 == 0) {
+                    delegate.setLocalNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+                } else {
+                    delegate.setLocalNightMode(AppCompatDelegate.MODE_NIGHT_YES)
+                }
+                Toast.makeText(this, "Change theme", Toast.LENGTH_SHORT).show()
+                true
+            }
+            R.id.menu_save_last_result -> {
+                val sharePreferences: SharedPreferences =
+                    getSharedPreferences("last result", Context.MODE_PRIVATE)
+                val editor: SharedPreferences.Editor = sharePreferences.edit()
+                editor.putFloat("last result", mResult)
```

Comment:
tương tự với tên giá trị trong shared preferences
tham khảo https://github.com/framgia/coding-standards/blob/master/vn/android/codingstyleguide.md#210-string-constants-naming-and-values